app/Dockerfile

```diff
@@ -1,14 +1,23 @@
-FROM ubuntu:14.04
+FROM debian:jessie
 
-RUN apt-get update && apt-get -y install curl netcat
-RUN mkdir -p /mattermost/data
+RUN apt-get update && apt-get -y install curl \
+ && rm -rf /var/lib/apt/lists/* \
+ && mkdir -p /opt/mattermost/data \
+ && useradd -r mattermost -U \
+ && cd /opt \
+ && curl https://releases.mattermost.com/3.0.2/mattermost-team-3.0.2-linux-amd64.tar.gz | tar -xvz
 
-RUN curl https://releases.mattermost.com/3.0.2/mattermost-team-3.0.2-linux-amd64.tar.gz | tar -xvz
+# grab gosu for easy step-down from root
+RUN gpg --keyserver pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.3/gosu-$(dpkg --print-architecture)" \
+ && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.3/gosu-$(dpkg --print-architecture).asc" \
+ && gpg --verify /usr/local/bin/gosu.asc \
+ && rm /usr/local/bin/gosu.asc \
+ && chmod +x /usr/local/bin/gosu
 
 COPY config.template.json /
-COPY docker-entry.sh /
+COPY docker-entrypoint.sh /entrypoint.sh
 
-RUN chmod +x /docker-entry.sh
-ENTRYPOINT ["/docker-entry.sh"]
+ENTRYPOINT ["/entrypoint.sh"]
 
-EXPOSE 80
+EXPOSE 8080
```

Comment:
Nice feature, I wasn't aware of. I still think that if you want to make it official, you should run it as non root. (as most user will not run docker with this special flag)

As I told you this PR, is a FYI, I let you decide to merge what you want.

If you want to make mattermost an official image, I can help you on this process, and I think non root is necessary.
If you don't want to make it official, you are free to take the security measures you want.